src/Featurizers/Components/StatisticsMetricsEstimator.h

```diff
@@ -0,0 +1,391 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "../../Traits.h"
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         StatisticalMetricsEstimatorName("StatisticalMetricsEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         BasicStatisticalAnnotationData
+///  \brief         An annotation class which contains the min, max and count
+///                 for an input column. Template T can be all types
+///
+template <typename T>
+class BasicStatisticalAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    T             const                           Min;
+    T             const                           Max;
+    std::uint64_t  const                          Count;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    BasicStatisticalAnnotationData(T min, T max, std::uint64_t count);
+    ~BasicStatisticalAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(BasicStatisticalAnnotationData);
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         StandardStatisticalAnnotaionData
+///  \brief         An annotation class which contains the sum and average
+///                 for an input column. Template T can only be numerical types
+///
+template <typename T>
+class StandardStatisticalAnnotaionData : public BasicStatisticalAnnotationData<T> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    long double   const                            Sum;

```

Comment:
I think that the Sum should be a type whose value is based on T. For example, Sum doesn't need to be a numeric type if T is an integer type.